### PR TITLE
D42 13677 - Dell API v5 Support

### PR DESCRIPTION
**Pull request by osanchez42**
added support for Dell Warranty API version 5. This new version uses OAuth2.0 structure for authentication so a new "client id" and "client secret" is required. Please update your API keys on Dell Tech Directs site to use this warranty script.

### CODE REVIEW

File: Files/warranty_dell.py [hunk 1]
```diff
@@ -68,6 +126,8 @@ def run_warranty_check(self, inline_serials, retry=True):
                     return None
             else:
                 result = resp.json()
+                print json.dumps(result)
```
Comment (cscaglioned42) [inline Files/warranty_dell.py]: Should these prints go in a "if debug" check?

File: Files/warranty_dell.py [hunk 2]
```diff
@@ -77,114 +137,101 @@ def process_result(self, result, purchases):
...
+            except IndexError:
+                if self.debug:
+                    try:
+                        msg = str(result['InvalidFormatAssets']['BadAssets'])
+                        if msg:
+                            print '\t\t[-] Error: Bad asset: %s' % msg
+                    except Exception as e:
+                        print e
 
+            else:
```
Comment (cscaglioned42) [inline Files/warranty_dell.py:150]: This "else" does not seem to be properly lined up with an "if".

File: Files/warranty_dell.py [hunk 3]
```diff
@@ -77,114 +137,101 @@ def process_result(self, result, purchases):
...
+                # saw this order number code, did not see this in the response with test devices,
+                # but leaving it here in case there is a product that does return this information
+                # if self.order_no == 'vendor':
```
Comment (cscaglioned42) [inline Files/warranty_dell.py]: Do we need these 2 commented out code lines.

File: Files/warranty_dell.py [hunk 4]
```diff
@@ -77,114 +137,101 @@ def process_result(self, result, purchases):
...
-                    for sub_item in warranties:
+                serial = item['serviceTag']
+
+                # We need check per warranty service item
+                for sub_item in warranties:
+                    data.clear()
+                    ship_date = item['shipDate'].split('T')[0]
+                    try:
+                        product_id = item['ProductId']
+                    except:
```
Comment (cscaglioned42) [inline Files/warranty_dell.py]: If you are only expecting a KeyError, make the except for that.

File: Files/warranty_dell.py (hunk 2 above)
Comment (osanchez42) [reply Files/warranty_dell.py:150]: this is a try-except-else clause. was confused about this too but it seems it is an alternative to finally that is run before all finally statements. Previous version was written like this

File: Files/warranty_dell.py (hunk 3 above)
Comment (osanchez42) [reply Files/warranty_dell.py]: i'll remove, I didn't see this value in the response. If we need to add them back in will do so

**Pull request merged**